app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupingMakingFragment.java

```diff
@@ -70,27 +79,62 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
         settings = new GroupingSettings(
+                listInfo.getNumNames(),
                 getResources().getInteger(R.integer.default_number_of_names_per_group),
                 getResources().getInteger(R.integer.default_number_of_groups));
         settingsDialog = new GroupingSettingsDialog(getActivity(), settings);
     }
 
+    @Override
+    public void onNameAdded(String name, int amount, String listName) {
+        listInfo = new DataSource(getContext()).getListInfo(listName);
+        settings.setListSize(listInfo.getNumNames());
```

Comment:
@Gear61 Should I put the getNumInstances() or getNumNames() the getNumNames() will not count the duplicated names and what we will do about this issue

Reply:
Good question! We should use getNumInstances(). We shouldn't flatten the names when there's multiple instances I think; that doesn't feel like expected behavior.